plugins/device/device.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package hashicorp.nomad.plugins.device;
+
+import "google/protobuf/empty.proto";
+
+// DevicePlugin is the API exposed by device plugins
+service DevicePlugin {
+  // FingerprintDevices allows the device plugin to return a set of   
+  // detected devices and provide a mechanism to update the state of
+  // the device.
+  rpc FingerprintDevices(google.protobuf.Empty) returns (stream DetectedDevices) {}
```

Comment:
Curious as to the reason you chose to do a stream here instead of batching everything inside a `FingerprintDevicesResponse` message.

Reply:
This allows the system to handle devices being added after initial fingerprinting

Reply:
Ah so it is forever running then?

Reply:
Yes!